### DISCUSSION TITLE
cann't use sdk on centos

### DISCUSSION THREAD

**Discussion by qingpingwang** [google/filament#6036] [General]
**My CmakeList**
```
cmake_minimum_required(VERSION` 3.21)

project(mogic_cg)

set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
set(CMAKE_CXX_STANDARD 17)

set(CXX_STANDARD "/std:c++latest")

##########  filament test  #############

set(FILAMENT_DIR ./third_party/filament/out/debug/filament)
set(FILAMENT_LIB_DIR ./third_party/filament/out/debug/filament/lib/x86_64)
file(GLOB FILAMENT_LIBS   ${FILAMENT_LIB_DIR}/libfilament.a ${FILAMENT_LIB_DIR}/libbackend.a ${FILAMENT_LIB_DIR}/libbluegl.a ${FILAMENT_LIB_DIR}/libbluevk.a ${FILAMENT_LIB_DIR}/libfilabridge.a ${FILAMENT_LIB_DIR}/libfilaflat.a ${FILAMENT_LIB_DIR}/libutils.a ${FILAMENT_LIB_DIR}/libgeometry.a ${FILAMENT_LIB_DIR}/libsmol-v.a ${FILAMENT_LIB_DIR}/libvkshaders.a ${FILAMENT_LIB_DIR}/libibl.a)

add_executable(filament_cg_demo samples/filament_cg_demo.cpp)
target_compile_options(filament_cg_demo PRIVATE -fpermissive -fno-builtin)
target_include_directories(filament_cg_demo PUBLIC ${FILAMENT_DIR}/include/)
target_link_directories(filament_cg_demo PUBLIC ${FILAMENT_DIR}/lib/x86_64/)
target_link_libraries(filament_cg_demo ${FILAMENT_LIBS} pthread c++ c dl)
```

** Get Error Message **
![image](https://user-images.githubusercontent.com/50129609/188111539-206d8c7c-950d-4ddd-9dd7-32833a522c22.png)

**To Reproduce**
1. compile filament with clang 11.1.0;
2. add this sample;
3. compile this sample using clang 11.1.0 : cmake ../ -DCMAKE_BUILD_TYPE=${build_type};

**Comment by qingpingwang** [comment]
Would this be the case without a GPU?
Whether I need to compile additional Swiftshader？

**Reply by dandingol03** [reply]
> Would this be the case without a GPU? Whether I need to compile additional Swiftshader？

Hi, i also want to use filament to do some rendering on centos.Have you got dockerfile? I found the prebuilt library which is needed for linux to compile filament is too much. It seems difficlut for me

**Comment by romainguy** [comment]
Looks like you are not linking against some of the libraries (particularly `libbackend.a`). I don't think you want to use `file(GLOB …)` here. Try to just `set(FILAMENT_LIBS …)` instead.
